src/router/validation/hapiFhirLambdaValidator.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { InvalidResourceError, Validator } from 'fhir-works-on-aws-interface';
+import { Lambda } from 'aws-sdk';
+import AWS from '../../AWS';
+
+interface ErrorMessage {
+    severity: string;
+    msg: string;
+}
+
+interface HapiValidatorResponse {
+    errorMessages: ErrorMessage[];
+    successful: boolean;
+}
+
+export default class HapiFhirLambdaValidator implements Validator {
+    private hapiValidatorLambdaArn: string;
+
+    private lambdaClient: Lambda;
+
+    constructor(hapiValidatorLambdaArn: string) {
+        this.hapiValidatorLambdaArn = hapiValidatorLambdaArn;
+        this.lambdaClient = new AWS.Lambda();
+    }
+
+    async validate(resource: any): Promise<void> {
+        const params = {
+            FunctionName: this.hapiValidatorLambdaArn,
+            InvocationType: 'RequestResponse',
+            Payload: JSON.stringify(resource),
+        };
+
+        const lambdaResponse = await this.lambdaClient.invoke(params).promise();
```

Comment:
Should we have a timeout here or does the lambda itself have a short time-out?

Reply:
It's probably fine to not set a client timeout here and just use the Lambda timeout. It is currently 1 minute but I'll lower it.

It's worth noting that, even with provisioned concurrency, there can be spikes of call volume that trigger cold starts, so we need to have a relatively high timeout value. It's probably better to allow a few requests to have latencies of >10 seconds due to cold starts than to have those requests explicitly fail due to a client timeout.

Reply:
What would be the error returned to the client if the validation takes longer than 30 secs (API GW timeout)?

Reply:
If API GW times out after 30 seconds. clients get:
```json
504 Gateway Timeout
{
    "message": "Endpoint request timed out"
}
```
Which is not so great since it is not wrapped in an `OperationOutcome` but this is existing behavour regardless of what the validator does.

I made a change to add a lambda client timeout. I'll still change the time out on CFN but it's ok to have the timeout in both places. If the lambda client times out, an exception is thrown which is not handled(there's nothing meaningful to do) and ends up as a 500 error with a proper `OperationOutcome` "Internal Server Error" response